File: etc/rc.d/restoreres.c
#include <spawn.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
extern char **environ;

int run_cmd(char *cmd)
{
    pid_t pid;
    char *argv[] = {"sh", "-c", cmd, NULL};
    int status = posix_spawn(&pid, "/bin/sh", NULL, NULL, argv, environ);
    if (status == 0) {
        if (waitpid(pid, &status, 0) == -1) {
            perror("waitpid");
        }
    }
    return status;
}

int main()
{
    if (geteuid() != 0) {
        printf("Run this as root!\n");
        exit(1);
    }
    
    if (access("/.cydia_no_stash", F_OK) != 0) {
        FILE *fp = fopen("/.firstboot","a+");
        fclose(fp);
        if (access("/var/mobile/Library/Preferences/com.michael.iokit.IOMobileGraphicsFamily.plist", F_OK) == 0) {
            remove("/var/mobile/Library/Preferences/com.apple.iokit.IOMobileGraphicsFamily.plist");
        }
    }
    return 0;
}

File: etc/rc.d/restoreres.m
int main()
{
    if (getuid() != 0) {
        printf("Run this as root!\n");
        return 1;
    }

    NSString *const userIOMobileGraphicsFamilyPlist = @"/var/mobile/Library/Preferences/com.michael.iokit.IOMobileGraphicsFamily.plist";
    NSDictionary *const userIOMobileGraphicsFamily = [NSDictionary dictionaryWithContentsOfFile:userIOMobileGraphicsFamilyPlist];
    if (userIOMobileGraphicsFamily[@"canvas_height"] && userIOMobileGraphicsFamily[@"canvas_width"]) {
        if (access("/var/mobile/Library/Preferences/com.apple.iokit.IOMobileGraphicsFamily.plist", F_OK) == 0){
            remove("/var/mobile/Library/Preferences/com.apple.iokit.IOMobileGraphicsFamily.plist");
        }
        if (access("/Library/MobileSubstrate/DynamicLibraries/AAAAAResSetter.disabled", F_OK) == 0 && access("/Library/MobileSubstrate/DynamicLibraries/AAAAAResSetter.dylib", F_OK) != 0){
            system("mv /Library/MobileSubstrate/DynamicLibraries/AAAAAResSetter.disabled /Library/MobileSubstrate/DynamicLibraries/AAAAAResSetter.dylib");
        }
    }
    return 0;
}
